dev/io.openliberty.microprofile.telemetry.2.0.logging.internal/src/io/openliberty/microprofile/telemetry20/logging/internal/MpTelemetryLogMappingUtils.java

```diff
@@ -394,8 +400,16 @@ else if (value instanceof Integer)
         // Set the Attributes to the builder.
         builder.setAllAttributes(attributes.build());
 
-        // Set the Span and Trace IDs from the current context.
-        builder.setContext(Context.current());
+        // Set the Span and Trace IDs from the current context. We're not on the same thread at the point when access logs are collected
+        // so we need to extract the trace/span ID from the 'traceparent' request header.
+        if (traceParent != null) {
+            String[] traceSplit = traceParent.split("-");
+
+            SpanContext customSpanContext = SpanContext.create(traceSplit[1], traceSplit[2], TraceFlags.getSampled(), TraceState.getDefault());
```

Comment:
Are we sure that the traceSplit, will always be immutable? Are we sure that the 1st and 2nd indices will always contain the info we need? We might need to add an unit test or some sort to test for this, just in case things change in future versions....

Reply:
Yes, this is the traceparent format: `00-0af7651916cd43dd8448eb211c80319c-b9c7c989f97918e1-01`

I've also added additional support for jaeger and b3, the other more popular configruable propagators.